### DISCUSSION TITLE
form.setValue is turning it into an array

### DISCUSSION THREAD

**Discussion by zyasir** [react-hook-form/react-hook-form#6682] [Q&A]
I am trying to do a `form.setValue('groups.800.cost')` such that I get this result:

```
groups: {
  '800': {
    cost: 10
  }
}
```

However I get this:

```
groups: [
,
,
,
,
.......
{ cost: 10 }
]
```

And my schema validation fails saying it expects object, not array.

Lodash works around this issue with `setWith` - https://lodash.com/docs/4.17.15#setWith

I also tried virutal registering the field first with:

```
form.register('groups', { value: {} })
```

But it doesn't help.

**Comment by thanh-nguyen-95** [comment]
This is quite an interesting edge case. I think the library assumes the string `'800'` is an index of an array and then convert `groups` to an Array.

I created a CodeSandBox to test the theory: https://codesandbox.io/s/determined-framework-k2tp5?file=/src/App.js

**Reply by bluebill1049** [reply]
yes, please avoid integer as the key name.
